woocommerce/woocommerce-ios
# [Mobile Payments] Consolidate CardPresent and PaymentGateway Stores; Switch Between WCPay and Stripe Backends Within

Closes: #5853
Closes: #5534
Closes: #5536
Closes: #5531

### Description
- Combines PaymentGatewayAccountStore and CardPresentPaymentStore, moving the actions in the former into the latter
- Consolidates their unit tests as well
- Changes onboarding to defer loading accounts until after it knows which plugin is active
- Also tells the CardPresentPaymentStore which backend to use for things like fetching accounts, connecting readers
- Defaults to WCPay backend

### Unfinished Business
- Define a protocol for the two remotes to simplify the code in CardPresentPaymentStore
- Add unit tests to cover the logic added to CardPresentPaymentStore and CardPresentPaymentsOnboardingUseCase - see also #5866 and #5867
- **I'm going to move these into subsequent PR(s) - this one is already so big.**

### Please note
- This does NOT hook up these changes to the Order flow. Capturing payments through Stripe powered stores will come in a later PR.

### Testing instructions
- First, ensure you can connect to a reader on a WCPay powered store
- Next, set up a store (if you haven't already) using the Stripe extension.
- IMPORTANT: Modify woocommerce-gateway-stripe/includes/admin/class-wc-rest-stripe-account-controller.php so that get_account_summary also returns keys of is_live (false) and test_mode (true) if necessary - see https://github.com/woocommerce/woocommerce-gateway-stripe/issues/2262 for more details
- Enable the Stripe IPP feature under Settings > Experimental
- Then, ensure you can connect to a reader on that Stripe powered store

### Screenshots
N/A

---
